zanata-war/src/main/java/org/zanata/service/impl/ActivityServiceImpl.java

```diff
@@ -195,9 +195,9 @@ public void logTextFlowStateUpdate(@Observes(during = TransactionPhase.AFTER_SUC
                 HDocument document =
                     documentDAO.getById(event.getKey().getDocumentId());
```

Comment:
To be on the safe side, I would move this back inside the transaction the way it used to be.